newhelm/runners/simple_benchmark_runner.py

```diff
@@ -96,16 +98,23 @@ def run_prompt_response_test(
         test_items = random.sample(test_items, max_test_items)
     item_interactions: List[TestItemInteractions] = []
     desc = f"Collecting responses to {test_name} from {sut_name}"
+    cache_helper = SUTResponseCacheHelper(
+        os.path.join(test_data_path, "cached_responses"), sut
+    )  # type: ignore
     for item in tqdm(test_items, desc=desc):
         interactions = []
         for prompt in item.prompts:
             sut_request = sut.translate_request(prompt.prompt)
-            sut_response = sut.evaluate(sut_request)
+            sut_response = cache_helper.get_cached_response(sut_request)
+            if sut_response is None:
+                sut_response = sut.evaluate(sut_request)
+                cache_helper.update_cache(sut_request, sut_response)
             response = sut.translate_response(prompt.prompt, sut_response)
             interactions.append(PromptInteraction(prompt=prompt, response=response))
         item_interactions.append(
             TestItemInteractions(interactions=interactions, test_item=item)
         )
+    cache_helper.close()
```

Comment:
How important is it that the cache close? Should we have SUTResponseCacheHelper implement `__enter__` and `__exit__` so then we have `with SUTResponseCacheHelper(...) as cache_helper:`?

Reply:
I'm honestly not sure. I can do it that way anyway just in case.